File: src/frontend/src/features/SummaryFeature.ts
import type { ProjectFeatureID } from "@common/data/entities/EntityTypes";
import { ProjectFeature, ProjectFeatureFlags } from "@common/features/ProjectFeature";

export class SummaryFeature extends ProjectFeature {
    public static readonly FeatureID: ProjectFeatureID = "summary";

    public constructor() {
        super(SummaryFeature.FeatureID, {
            flags: ProjectFeatureFlags.HasPanel,
            displayName: "Summary",
            panel: () => import("@/ui/content/features/SummaryPanel.vue")
        });
    }
}

File: src/common/py/features/summary_feature.py
from .project_feature import ProjectFeature


class SummaryFeature(ProjectFeature):
    FEATURE_ID = "summary"

    def __init__(self):
        super().__init__(
            feature_id=SummaryFeature.FEATURE_ID,
            flags=ProjectFeature.FeatureFlags.NONE,
            display_name="Summary",
        )
